napari/components/layerlist.py

```diff
@@ -211,6 +226,63 @@ def extent(self) -> Extent:
             step=self._get_step_size(extent_list),
         )
 
+    @property
+    def _ranges(self) -> List[Tuple[float, float, float]]:
+        """Get ranges for Dims.range in world coordinates.
+
+        This shares some code in common with the `extent` property, but
+        determines Dims.range settings for each dimension such that each
+        range is aligned to pixel centers at the finest scale.
+        """
+        if len(self) == 0:
+            return [(0, 1, 1)] * self.ndim
+        else:
+            # Determine minimum step size across all layers
+            layer_extent_list = [layer.extent for layer in self]
+            scales = [extent.step for extent in layer_extent_list]
+            min_steps = self._step_size_from_scales(scales)
+
+            # Pixel-based layers need to be offset by 0.5 * min_steps to align
+            # Dims.range with pixel centers in world coordinates
+            pixel_offsets = [
+                0.5 * min_steps
+                if isinstance(layer, _ImageBase)
+                else [0] * len(min_steps)
+                for layer in self
+            ]
+
+            # Non-pixel layers need the an offset of the range stop by min_steps since the upper
+            # limit of Dims.range is non-inclusive.
+            point_offsets = [
+                [0] * len(min_steps)
+                if isinstance(layer, _ImageBase)
+                else min_steps
+                for layer in self
+            ]
+
+            # Determine world coordinate extents similarly to
+            # `_get_extent_world`, but including offsets calculated above.
+            extrema = [extent.world for extent in layer_extent_list]
+            mins = [
+                e[0] + o1[: len(e[0])] for e, o1 in zip(extrema, pixel_offsets)
+            ]
+            maxs = [
+                e[1] + o1[: len(e[0])] + o2[: len(e[0])]
+                for e, o1, o2 in zip(extrema, pixel_offsets, point_offsets)
+            ]
+            min_v, max_v = self._get_min_and_max(mins, maxs)
+            # same 512 element default extent as `_get_extent_world`
+            min_v = np.asarray(min_v)
+            min_v[np.isnan(min_v)] = -0.5
+            max_v = np.asarray(max_v)
+            max_v[np.isnan(max_v)] = 511.5
```

Comment:
As above, this can be simplified if we move to numpy 1.18+, as we should, and also fold it into _get_min_and_maxs